Adding CPG Guidelines for MDN (#38272)
mdn/content: files/en-us/mdn/community/community_participation_guidelines/index.md

---
title: Community Participation Guidelines
slug: MDN/Community/Community_Participation_Guidelines
page-type: mdn-community-guide
sidebar: mdnsidebar
---

### Purpose

> The Community Participation Guidelines (CPG) exist to foster a welcoming, inclusive, and safe community. This document outlines clear processes for reporting violations and the actions that will be taken by the enforcement team. Our focus is to ensure that every community member feels valued, heard, and protected.

Reviewer: I changed it to "these" because I thought this document was the CPG based on the title and slug, but these are the consequences for code of conduct violations, not the guidelines themselves. This page does not link to those either.

> The Community Participation Guidelines (CPG) exist to foster a welcoming, inclusive, and safe community. This document outlines clear processes for reporting violations and the actions that will be taken by the enforcement team. Our focus is to ensure that every community member feels valued, heard, and protected.

Author: I've updated these to improve clarity. I'd like for your to review it and share any improvements. Resolving this comment for now.

### Guiding Principles

- **Transparency:** Every report is addressed clearly and consistently.
- **Confidentiality:** Reporters' privacy will be protected unless explicit consent is given to share information.
- **Proportionality:** The response to violations will be proportionate to the harm caused.

> - **Restorative Justice:** Where possible, the process aims to repair harm and foster learning.

Author: Looks good, thanks!

## Code of Conduct Violations and Enforcement Ladder

> The CPG outlines several categories of violations. These are summarized below along with appropriate enforcement actions. The enforcement process follows an escalation ladder, where individuals are made aware of future consequences each time an action is taken. The ladder is not strictly linear—depending on the severity of the violation, escalation may skip levels or result in an immediate ban.

Reviewer: ```suggestion
The CPG outlines several categories of violations. These are summarized below along with appropriate enforcement actions. The enforcement process follows an escalation ladder, where individuals are made aware of future consequences each time an action is taken. The ladder may not be strictly linear— depending on the severity of the violation, escalation may skip levels up to and including immediate, permanent bans.
```

### Level 0: No Action

> - **Description:** Recommendations do not indicate a violation of the CPG.

Reviewer: ```suggestion
- **Description:** No indication of community guideline violations.
```
not sure what "recommendations" mean here, so my suggestion may be off.

Author: The term "recommendations" here refers to suggestions or advice offered by community members that, in and of themselves, do not constitute a violation of the CPG.

Author: This has been changed, thanks Estelle! 🙌

> - **Actions:** No formal action taken, but feedback may be provided to promote positive behavior.

Reviewer: ```suggestion
- **Actions:** No formal action taken. Feedback may be provided to the reporter, reportee, or both, to promote good behavior and positive interactions.
```

Reviewer: sometimes it is the reporter that violated the CoC; especially when the reportee didn't violate anything, which may be targeted harassment.

Author: That's an important point, thanks Estelle!

### Level 1: Simple Warning Issued

- **Description:** Minor violations of the CPG.

> - **Actions:** A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.

Reviewer: this is usually from a contributor, not leadership, so suggest
```suggestion
- **Actions:** A private, written warning from a team member, with clarity of the violation and consequences of continued behavior.
```

Author: Updated language for project leadership → MDN Community management

### Level 2: Warning

> - **Description:** Repeated minor violations or a single moderate violation of the CPG.

Reviewer: I suggest putting the "first offense gets this level" first, followed by the repeat offense, in this case and all others.
```suggestion
- **Description:** Moderate violation of the CPG, or repeated minor violations.
```

- **Actions:**

>   - A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.

Reviewer: ```suggestion
  - A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.
  - Guidance on rectification of aggrieving situation, if applicable.
```

  - Communication of next-level consequences if behaviors are repeated.

### Level 3: Warning + Mandatory Cooling Off Period (Tool Access Revoked - GH & Discord)

- **Description:** Continued violations or escalation in severity.
- **Actions:**
  - A private warning from project leadership, with clarity of the violation and consequences of continued behavior.
  - Request to avoid interaction on community messaging platforms (public forums, commenting on bugs).
  - Required to not interact with others involved in the report or those suspected to be involved.
  - This includes avoiding any interactions in Mozilla channels, spaces/offices, and external channels like social media (e.g., following/liking/retweeting).

> Mandatory cooling-off period. </br>
> Mandatory CPG onboarding process.

### Level 4: Ban across MDN and Mozilla Spaces, All Areas Permanent Ban

- **Description:** Permanent ban from all Mozilla activities and spaces
- **Actions:**

>   - Private communication of ban from project leadership, with clarity of the violation and consequences of continued behavior.

Reviewer: There is no continued behavior as they are banned, so perhaps the following... though this is more like a level 5 suggestion
```suggestion
  - Private communication of ban from project leadership, with clarity of the violation and warning of consequences should the behavior continue off of Mozilla properties.
```

Author: Hi @estelle, I've changed this. Let me know if this works.

  - Required to avoid interaction on community messaging platforms and external channels.

>   - All Mozilla accounts will be deactivated permanently.
>   - Not allowed to enter Mozilla digital and event spaces permanently.
>   - All community leadership roles will be permanently suspended.
>   - Permission to use the Mozilla trademark is revoked.

Reviewer: starting with verbs
```suggestion
  - Deactivation of any and all Mozilla.
  - Permanently banned from entering Mozilla digital and event spaces.
  - Permanently suspension from any and all community leadership roles.
  - Revocation of any and all permissions to use the Mozilla trademark.
```

Reviewer 2: Minor ones:

>any and all Mozilla accounts.

>Permanently suspended

Author: Edited, thanks both! 🙌

## Reporting Process

### How to Report

> Reports can be submitted via mdn-cpg-report@mozilla.com. The report should include all relevant details (e.g., date, time, description, screenshots).

Reviewer: ```suggestion
Violations of community participation guidelines should be sent to mdn-cpg-report@mozilla.com. Please include all relevant details, including the date, time, and description of the violation. Include identifying information of the offender and screenshots, if possible.
```

## Code of Conduct Violations and Enforcement Ladder

[MDN Community Management](https://github.com/orgs/mdn/teams/community) investigates the severity of behavior based on a report, then determines and carries out any actions, if appropriate.
Violations follow a step-by-step escalation process, and the reportee is informed of consequences at leach level.
In serious cases, enforcement may skip levels up to and including an immediate, permanent ban.

Below is a summary of severity levels and enforcement actions.

### Level 0: No Action

- **Severity:** There is no clear indication a violation of the CPG.
- **Actions:** No formal action taken. Feedback is provided to the reporter, reportee, or both, to promote good behavior and positive interactions.

### Level 1: Simple Warning Issued

- **Severity:** Minor violations of the CPG.
- **Actions:** A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.

### Level 2: Warning

- **Severity:** Moderate violation of the CPG, or repeated minor violations.
- **Actions:**
  - A private, written warning with clarity of the violation and consequences of continued behavior.

>   - Guidance on rectification of aggrieving situation, if applicable.

Reviewer 2: ```suggestion
  - Guidance on rectifying the situation, if possible.
```

  - Avoid interactions in Mozilla channels, physical spaces and offices, and online channels like social media (e.g., following/liking/retweeting).
  - The reportee has the opportunity to participate in the project again after onboarding that clarifies expected behavior.

### Level 4: Ban across MDN and Mozilla Spaces, All Areas Permanent Ban

- **Severity:** Serious [Mozilla Community Participation Guidelines (CPG)](https://www.mozilla.org/en-US/about/governance/policies/participation/) breaches
- **Actions:**
  - Private communication of a ban from project leadership, with clarity of the violation and warning of consequences. Should the behavior continue, it would lead to a ban from all Mozilla community spaces.
  - Required to avoid interaction on community messaging platforms and external channels.
  - Deactivation of any and all Mozilla accounts.
  - Permanently banned from entering Mozilla digital and event spaces.
  - Permanently suspended from any and all community leadership roles.
  - Revocation of any and all permissions to use the Mozilla trademark.

## Reporting Process

### How to Report a violation

> Reports should be sent to mdn-cpg-report@mozilla.com.

Reviewer 2: ```suggestion
Reports can be sent to mdn-cpg-report@mozilla.com.
Content flagged on MDN GitHub repositories is visible to MDN staff when the "Report to repository admins" option is checked, and this may be used to report CPG violations.
See GitHub's [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam) documentation for details.
```

---

> title: Community Participation Guidelines (CPG)

Reviewer: ```suggestion
title: Code of conduct enforcement guidelines 
```
This document is about reporting and enforcement. The CPG is on moz.org

Reviewer 2: >This document is about reporting and enforcement.

That's true. I like that we mention CPG in the title, seeing as that's [the CoC](https://github.com/mdn/content/blob/main/CODE_OF_CONDUCT.md):

> CPG reports and enforcement

or even this is another one, though it's getting long:

>Code of conduct (CPG) reports and enforcement

Reviewer 2: Actually I think the original is also good, there's some intro to the guidelines, the process, then FAQs, so it's not just reporting. What do you think?

Reviewer: CPG is a mozilla acronym/name. Everyplace else calls it CoC. I am thinking "Code of conduct (CPG) reports and enforcement", which includes both - what people will be looking for but what we call it on MDN - is the best.

Reviewer 2: Looks good to me 👍

In serious cases, enforcement may skip levels up to and including an immediate, permanent ban.

Below is a summary of severity levels and enforcement actions.

### Level 0: No Action

- **Severity:** There is no clear indication a violation of the CPG.
- **Actions:** No formal action taken. Feedback is provided to the reporter, reportee, or both, to promote good behavior and positive interactions.

### Level 1: Simple Warning Issued

- **Severity:** Minor violations of the CPG.
- **Actions:** A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.

### Level 2: Warning

- **Severity:** Moderate violation of the CPG, or repeated minor violations.
- **Actions:**
  - A private, written warning with clarity of the violation and consequences of continued behavior.
  - Guidance on rectification of aggrieving situation, if applicable.
  - Communication of next-level consequences if behaviors are repeated.

### Level 3: Warning + Mandatory Cooling Off Period (Tool Access Revoked - GH & Discord)

- **Severity:** Continued violations or escalation in severity.
- **Actions:**

>   - A private warning with clarity of the violation and consequences of continued behavior.
>   - Request to avoid interaction on community messaging platforms (public forums, commenting on bugs).
>   - Required to not interact with others involved in the report or those suspected to be involved.
>   - This includes avoiding any interactions in Mozilla channels, spaces/offices, and external channels like social media (e.g., following/liking/retweeting).
> 
> > Mandatory cooling-off period.
> >
> > Mandatory CPG onboarding process.

Reviewer 2: Let's put it into the bullet points under "actions", because it looks lonely out there by itself as a quote block and it's hard to know what it means

```suggestion
    - A private warning with clarity of the violation and consequences of continued behavior.
    - **Mandatory cooling-off period:**
    - No interaction on community messaging platforms, such as public forums, commenting on bugs.
    - No interacting with people involved in the report or others suspected to be involved.
    - Avoid interactions in Mozilla channels, physical spaces and offices, and online channels like social media (e.g., following/liking/retweeting).
  - **Mandatory CPG onboarding process.**
    The reportee has the opportunity to participate in the project again after onboarding that clarifies expected behavior.
```

---
title: Community Participation Guidelines (CPG)
short-title: Community Participation Guidelines
slug: MDN/Community/Community_Participation_Guidelines
page-type: mdn-community-guide
sidebar: mdnsidebar
---

### Purpose

> 

Reviewer 2: Reason being the first section heading should be a H2 (`## thing`) and in this case "Purpose" doesn't add much over the intro para.

> 

Author: Makes sense.

Participating in MDN Web Docs projects means adhering to the [Mozilla Community Participation Guidelines (CPG)](https://www.mozilla.org/en-US/about/governance/policies/participation/). These guidelines exist to foster a welcoming, inclusive, and safe environment, and to ensure that every community member feels valued, heard, and protected.

This document describes how to report violations and what happens depending on the severity of a violation.

> ### Guiding Principles

Author: Looks good!

- **Transparency:** Address every report clearly and consistently.
- **Confidentiality:** Protect reporter privacy unless explicit consent is given to share information.
- **Proportionality:** Responses to violations are proportionate to the harm caused.
- **Restorative Justice:** Repair harm and foster learning.

## Code of Conduct Violations and Enforcement Ladder

MDN Community Management investigates the severity of behavior based on a report, then determines and carries out any actions, if appropriate.
Violations follow a step-by-step escalation process, and the reportee is informed of consequences at leach level.
In serious cases, enforcement may skip levels up to and including an immediate, permanent ban.

Below is a summary of severity levels and enforcement actions.

### Level 0: No Action

- **Severity:** There is no clear indication a violation of the CPG.
- **Actions:** No formal action taken. Feedback is provided to the reporter, reportee, or both, to promote good behavior and positive interactions.

> ### Level 1: Simple Warning Issued

Author: We do two levels of warning because the action is built on restorative justice, we want to repair harm and help folks improve.

- **Severity:** Minor violations of the CPG.
- **Actions:** A private, written warning from project leadership, with clarity of the violation and consequences of continued behavior.

### Level 2: Warning

- **Severity:** Moderate violation of the CPG, or repeated minor violations.
- **Actions:**
  - A private, written warning with clarity of the violation and consequences of continued behavior.
  - Guidance on rectification of aggrieving situation, if applicable.
  - Communication of next-level consequences if behaviors are repeated.

### Level 3: Warning + Mandatory Cooling Off Period (Tool Access Revoked - GH & Discord)

- **Severity:** Continued violations or escalation in severity.
- **Actions:**
  - A private warning with clarity of the violation and consequences of continued behavior.
  - Request to avoid interaction on community messaging platforms (public forums, commenting on bugs).
  - Required to not interact with others involved in the report or those suspected to be involved.
  - This includes avoiding any interactions in Mozilla channels, spaces/offices, and external channels like social media (e.g., following/liking/retweeting).

> Mandatory cooling-off period.
>

> > Mandatory CPG onboarding process.

Author: This is a note to share that there will be a mandatory cooling-off period and a CPG process to get back within the community.

> > Mandatory CPG onboarding process.

Author: Removed, thanks Brian! 🙌

### Level 4: Ban across MDN and Mozilla Spaces, All Areas Permanent Ban

- **Severity:** Serious [Mozilla Community Participation Guidelines (CPG)](https://www.mozilla.org/en-US/about/governance/policies/participation/) breaches
- **Actions:**
  - Private communication of ban from project leadership, with clarity of the violation and consequences of continued behavior.
  - Required to avoid interaction on community messaging platforms and external channels.
  - All Mozilla accounts will be deactivated permanently.
  - Not allowed to enter Mozilla digital and event spaces permanently.
  - All community leadership roles will be permanently suspended.
  - Permission to use the Mozilla trademark is revoked.

## Reporting Process

### How to Report a violation

> Reports should be sent to mdn-cpg-report@mozilla.com.

Reviewer 2: One other thing; we don't mention reporting on GitHub (which is a common one) and then what to do on other channels (Discord).

Reviewer 2: Aren't things reported on GitHub assumed to be CoC violations? A lot of time this turns out to be bot activity that doesn't need follow-ups, but this is a reporting mechanism, isn't it?

Author: That's true, most of the time they are a CoC violation for bot activity but any reported content will be escalated to a CPG investigation if warranted. I'll update this!

Author: Added the following; let me know if this works.
`To report a violation or abuse on GitHub, you can [use this guide](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam) for the MDN team to take necessary action.`

Include any relevant details, including date, time, description of the violation.
Include identifying information of the offender and screenshots, if possible.

### Investigation

MDN Community Management investigate all reports confidentially.
When necessary, further information may be requested from the involved parties.

### Decision Making

MDN Community Management assess the severity of the violation, apply the enforcement actions as outlined, and communicate the outcome to relevant parties.

### Appeals

If a reported individual believes the decision was unjust, they may appeal through the [appeals process detailed on the community page].
Appeals are reviewed by a separate, impartial panel.

### Documentation and Record Keeping

All incidents will be documented to ensure consistency and fairness.
Records of violations and actions taken are kept private, and accessible only by MDN Community Management.

> ## Frequently Asked Questions (FAQs)

Reviewer 2: We mention "MDN Community Management" in a few places, but we don't say who that is and how to contact them 😉 

There's also https://github.com/orgs/mdn/teams/mdn-community-engagement, it would be nice to be able to tag a team.

Author: Linked the team. Resolving this conversation! ✅
